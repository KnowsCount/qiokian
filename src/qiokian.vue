<!--
 * @Date: 10/02/2021 20.07.50 +0800
 * @Author: KnowsCount
 * @LastEditTime: 11/06/2021 22.41.21 +0800
 * @FilePath: /qiokian/src/qiokian.vue
-->
<template>
	<div></div>
</template>

<script>
export default {
	data() {
		return {
			live2d_path:
				'https://cdn.jsdelivr.net/gh/knowscount/live2d-widget@latest/',
			cdnPath: 'https://cdn.jsdelivr.net/gh/fghrsh/live2d_api/',
		}
	},
	mounted() {
		this.__init()
	},
	methods: {
		__init() {
			this.loadAssets()
		},
		loadAssets() {
			// 加载 waifu.css live2d.min.js waifu-tips.js
			if (screen.width >= 768) {
				Promise.all([
					this.loadExternalResource(
						this.live2d_path + 'waifu.css',
						'css'
					),
					this.loadExternalResource(
						this.live2d_path + 'live2d.min.js',
						'js'
					),
					this.loadExternalResource(
						this.live2d_path + 'waifu-tips.js',
						'js'
					),
				]).then(() => {
					initWidget({
						waifuPath: this.live2d_path + 'waifu-tips.json',
						//apiPath: "https://live2d.fghrsh.net/api/",
						cdnPath: this.cdnPath,
					})
				})
			}
			// initWidget 第一个参数为 waifu-tips.json 的路径，第二个参数为 API 地址
			// API 后端可自行搭建，参考 https://github.com/fghrsh/live2d_api
			// 初始化看板娘会自动加载指定目录下的 waifu-tips.json
		},

		// 封装异步加载资源的方法
		loadExternalResource(url, type) {
			// 注意：live2d_path 参数应使用绝对路径
			// const live2d_path =
			//   "https://cdn.jsdelivr.net/gh/stevenjoezhang/live2d-widget@latest/";
			//const live2d_path = "/live2d-widget/";
			return new Promise((resolve, reject) => {
				let tag

				if (type === 'css') {
					tag = document.createElement('link')
					tag.rel = 'stylesheet'
					tag.href = url
				} else if (type === 'js') {
					tag = document.createElement('script')
					tag.src = url
				}
				if (tag) {
					tag.onload = () => resolve(url)
					tag.onerror = () => reject(url)
					document.head.appendChild(tag)
				}
			})
		},
	},
}
</script>