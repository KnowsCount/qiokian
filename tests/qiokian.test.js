/*
 * @Date: 11/07/2016 16.00.01 +0800
 * @Author: KnowsCount
 * @LastEditTime: 28/06/2021 20.13.38 +0800
 * @FilePath: /qiokian/tests/qiokian.test.js
 */

/* eslint-disable no-undef */

import qiokian from '../src/qiokian.vue'
import { shallowMount } from '@vue/test-utils'
import assert from 'assert'

const wrapper = shallowMount(qiokian)

describe('qiokian', function() {
	it('should export an object', function() {
		assert(qiokian)
		assert.strictEqual(typeof qiokian, 'object')
	})

	it('should throw an error when invalid args are passed', function() {
		assert.throws(function() {
			qiokian()
		})
	})

	it('should contain a <button> & <div> element', () => {
		expect(wrapper.find('button')).toBeTruthy()
		expect(wrapper.find('div')).toBeTruthy()
		wrapper.destroy()
	})

	it("renders correctly", () => {
		expect(wrapper.html()).toMatchSnapshot()
	});

	// if (screen.width < 768) {
	// 	expect(wrapper.find('button')).toBeFalsy()
	// }
	// what happened to this part?... it says `Received: {"selector": "button"}`
})
