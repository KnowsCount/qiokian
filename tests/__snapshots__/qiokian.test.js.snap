// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`qiokian renders correctly 1`] = `"<div></div>"`;
